### ISSUE TITLE
`no-useless-fallback-in-spread`: Support ternary

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#1861]
The rule currently does not detect ternary like this:

```js
{...(foo ? {bar: true} : {})}
```

This can be safely be reduced to

```js
{...(foo && {bar: true})}
```

**Comment by fisker** [comment]
Nice catch, another one.

```js
{...(!foo ? {} : {bar: true})}
```

**Comment by fisker** [comment]
If it's

```js
({...(foo ? foo : {})})
```

It will be caught by [prefer-logical-operator-over-ternary](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-logical-operator-over-ternary.md), and fix to

```js
({...(foo ?? {})})
// or
({...(foo || {})}) 
```

Then `no-useless-fallback-in-spread` will report.

**Comment by silverwind** [comment]
When updating docs, we should remove the `const object =` parts as these are not relevant to the rule. Make it a simple expression example :)

**Comment by fisker** [comment]
Added `const object =` because object expression  need be parenthesized, otherwise it will be invalid, add `const object =` makes it more like a real world case.
